sys/net/sock/sock_util.c

```diff
@@ -251,6 +262,10 @@ int sock_tl_str2ep(struct _sock_tl_ep *ep_out, const char *str)
 #ifdef SOCK_HAS_IPV6
     if (inet_pton(AF_INET6, hostbuf, ep_out->addr.ipv6) == 1) {
         ep_out->family = AF_INET6;
+        if (*hostend != '%' &&
+            ipv6_addr_is_link_local((ipv6_addr_t *)ep_out->addr.ipv6)) {
+            return _get_default_netif(ep_out);
```

Comment:
Seems like the unittest never triggers this line, so we need an additional test case.

Reply:
I added one.